tests/core/string/test_string.h

```diff
@@ -456,30 +456,84 @@ TEST_CASE("[String] Number to string") {
 }
 
 TEST_CASE("[String] String to integer") {
-	static const char *nums[4] = { "1237461283", "- 22", "0", " - 1123412" };
-	static const int num[4] = { 1237461283, -22, 0, -1123412 };
+	static const char *nums[13] = { "1237461283", "- 22", "0", " - 1123412", "", "10_000_000", "-1_2_3_4", "10__000", "  1  2  34 ", "-0", "007", "0b1011", "0x1012" };
+	static const int num[13] = { 1237461283, -22, 0, -1123412, 0, 10000000, -1234, 10000, 1234, 0, 7, 1011, 1012 };
```

Comment:
```suggestion
	static const int num[13] = { 1237461283, -22, 0, -1123412, 0, 10000000, -1234, 10000, 1234, 0, 7, 0b1011, 0x1012 };
```
These values are not aligned, `1011` in decimal is not the same as `1011` in binary, same for hexadecimal `1012` and decimal `1012`

Reply:
`"0b1011"` and `"0x1012"` just look like a binary and a hexadecimal number, but the `to_int()` doesn't handle them as such. `to_int()` ignores any non digits, so what it sees is `"01011"` and `"01012"` and converts them as base-10.

Reply:
My bad, good point, that should be noted I think in the tests to clarify it's correct behavior